plugin-mainmenu/actionview.cpp

```diff
@@ -275,6 +281,38 @@ QSize ActionView::minimumSizeHint() const
     return QSize{0, 0};
 }
 
+void ActionView::mousePressEvent(QMouseEvent* event)
+{
+    if (event->button() == Qt::LeftButton)
+        mDragStartPosition = event->pos();
+
+    QListView::mousePressEvent(event);
+}
+
+void ActionView::mouseMoveEvent(QMouseEvent *event)
+{
+    if (!(event->buttons() & Qt::LeftButton))
+        return;
+
+    if ((event->pos() - mDragStartPosition).manhattanLength() < QApplication::startDragDistance())
+        return;
+
+    XdgAction *a = qobject_cast<XdgAction*>(indexAt(mDragStartPosition).data(ActionView::ActionRole).value<QAction*>());
+    if (!a)
+        return;
+
+    QList<QUrl> urls;
+    urls << QUrl::fromLocalFile(a->desktopFile().fileName());
+
+    QMimeData *mimeData = new QMimeData();
+    mimeData->setUrls(urls);
+
+    QDrag *drag = new QDrag(this);
+    drag->setMimeData(mimeData);
+    drag->exec(Qt::CopyAction | Qt::LinkAction);
+    emit requestShowHideMenu();
```

Comment:
For the sake of certainty, I would use `QPointer<QDrag>` and add this after emitting `requestShowHideMenu()`:

```c++
    event->accept();
    drag->deleteLater();
```

However, I don't think it would make a difference here. It's done in a few other parts of the code as you did it.

Reply:
done, but i just thought of something, if i use `QDrag drag(this);` without a pointer, it wouldn't require `deleteLater()` nor `QPointer` right?

Reply:
[Qt doc](https://doc.qt.io/qt-5/qdrag.html) says, "The QDrag must be constructed on the heap with a parent QObject to ensure that Qt can clean up after the drag and drop operation has been completed."